app/uk/gov/hmrc/soletraderidentificationfrontend/services/SummaryListRowBuilderService.scala

```diff
@@ -0,0 +1,137 @@
+      actions = Some(Actions(items = Seq(
+        ActionItem(
+          href = routes.CaptureFullNameController.show(journeyId).url,
+          content = Text("Change"),
+          visuallyHiddenText = Some(messages("check-your-answers.last_name"))
+        )
+      )))
+    )
+
+    val dobRow = SummaryListRow(
+      key = Key(content = Text(messages("check-your-answers.dob"))),
+      value = Value(content = Text(individualDetails.dateOfBirth.format(checkYourAnswersFormat))),
+      actions = Some(Actions(items = Seq(
+        ActionItem(
+          href = routes.CaptureDateOfBirthController.show(journeyId).url,
+          content = Text("Change"),
+          visuallyHiddenText = Some(messages("check-your-answers.dob"))
+        )
+      )))
+    )
+
+    val ninoRow = SummaryListRow(
+      key = Key(content = Text(messages("check-your-answers.nino"))),
+      value = Value(content = Text(
+        individualDetails.optNino match {
+          case Some(nino) => nino.grouped(2).mkString(" ")
+          case None => messages("check-your-answers.no_nino")
+        }
+      )),
+      actions = Some(Actions(items = Seq(
+        ActionItem(
+          href = routes.CaptureNinoController.show(journeyId).url,
+          content = Text("Change"),
+          visuallyHiddenText = Some(messages("check-your-answers.nino"))
+        )
+      )))
+    )
+
+    val sautrRow = SummaryListRow(
+      key = Key(content = Text(messages("check-your-answers.sautr"))),
+      value = Value(content = Text(
+        individualDetails.optSautr match {
+          case Some(utr) => utr
+          case None => messages("check-your-answers.no_sautr")
+        }
+      )),
+      actions = Some(Actions(items = Seq(
+        ActionItem(
+          href = routes.CaptureSautrController.show(journeyId).url,
+          content = Text("Change"),
+          visuallyHiddenText = Some(messages("check-your-answers.sautr"))
+        )
+      )))
+    )
+
+    val addressRowSeq = optAddress match {
+      case Some(address) =>
+        Seq(SummaryListRow(
+          key = Key(content = Text(messages("check-your-answers.home_address"))),
+          value = Value(HtmlContent(address.toCheckYourAnswersFormat(config.countries))),
+          actions = Some(Actions(items = Seq(
+            ActionItem(
+              href = routes.CaptureAddressController.show(journeyId).url,
+              content = Text("Change"),
+              visuallyHiddenText = Some(messages("check-your-answers.home_address"))
+            )
+          )))
+        ))
+      case _ => Seq.empty
+    }
+
+    if (optAddress.isDefined && enableSautrCheck && individualDetails.optNino.isEmpty)
+      Future.successful(Seq(firstNameRow, lastNameRow, dobRow, ninoRow, sautrRow) ++ addressRowSeq)
+    else if (enableSautrCheck)
+      Future.successful(Seq(firstNameRow, lastNameRow, dobRow, ninoRow, sautrRow))
+    else
+      Future.successful(Seq(firstNameRow, lastNameRow, dobRow, ninoRow))
```

Comment:
It might be better to have all of this logic contained within the rows themselves rather than having two points of logic.